src/utils/api-request.ts

```diff
@@ -111,12 +111,9 @@ class DefaultHttpResponse implements HttpResponse {
 }
 
 export class HttpError extends Error {
-
-  public readonly response: HttpResponse;
-
-  constructor(resp: LowLevelResponse) {
-    super(`Server responded with status ${resp.status}.`);
-    this.response = new DefaultHttpResponse(resp);
+  constructor(public readonly response: HttpResponse) {
+    super(`Server responded with status ${response.status}.`);
+    Object.setPrototypeOf(this, HttpError.prototype);
```

Comment:
What does this actually do? You are setting prototype of HttpError to prototype of HttpError?

Reply:
This is needed for the `instanceof` checks to work correctly: https://github.com/Microsoft/TypeScript/issues/13965

Reply:
Can you add a comment on this. There is no way I would have guessed this.